include/aikido/perception/VoxelGridModule.hpp

```diff
@@ -0,0 +1,84 @@
+#ifndef AIKIDO_PERCEPTION_VOXELGRIDMODULE_HPP_
+#define AIKIDO_PERCEPTION_VOXELGRIDMODULE_HPP_
+
+#include <memory>
+#include <string>
+
+#include <dart/dynamics/dynamics.hpp>
+#include <ros/ros.h>
+
+namespace aikido {
+namespace perception {
+
+/// Perception module to take point cloud ROS msg and update voxel grid.
+class VoxelGridPerceptionModule
+{
+public:
+  /// Constructor
+  ///
+  /// \param[in] nodeHandle ROS node handle.
+  /// \param[in] pointCloudTopic The topic name.
+  /// \param[in] voxelGridShape Voxel grid to update.
+  VoxelGridPerceptionModule(
+      const ros::NodeHandle& nodeHandle,
+      const std::string& pointCloudTopic,
+      std::shared_ptr<dart::dynamics::VoxelGridShape> voxelGridShape = nullptr);
+
+  /// Destructor
+  virtual ~VoxelGridPerceptionModule() = default;
+
+  /// Returns DART VoxelGridShape.
+  std::shared_ptr<dart::dynamics::VoxelGridShape> getVoxelGridShape();
+
+  /// Returns const DART VoxelGridShape.
+  std::shared_ptr<const dart::dynamics::VoxelGridShape> getVoxelGridShape()
+      const;
+
+  /// Receives ROS msg and updates the voxel grid with the received ROS msg.
+  ///
+  /// \param[in] sensorOrigin Origin of sensor relative to frame.
+  /// \param[in] inCoordinatesOf Reference frame, determines transform to be
+  /// applied to point cloud and sensor origin.
+  /// \param[in] timestamp Only detections more recent than this timestamp will
```

Comment:
This isn't in the function signature, so the description should either be deleted or the `timestamp` parameter should be added back.

Reply:
Oh, right. It should be `timeout`, and also the docstring should be updated accordingly.